pkg/database/sql/connection_test.go

```diff
@@ -0,0 +1,96 @@
+// Copyright KubeArchive Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package sql
+
+import (
+	"strings"
+	"testing"
+)
+
+func TestExtractConnectionInfo(t *testing.T) {
```

Comment:
Take a look at the passwords on our production clusters and test with variants that contain similar characters. `testpass` is fine, but what about `%123=HEllo&27AZ34!` encoded to `%25123%3DHEllo%2627AZ34%21`?